src/lib.rs

```diff
@@ -3,16 +3,93 @@
 #![allow(dead_code)]
 #![allow(clippy::identity_op)]
 
+//! # Library for MCP2517FD CAN controller
+//!
+//! Crate currently offers the following features:
+//! * CAN2.0 and CAN FD format support
+//! * Standard and extended ID formats for CAN frames
+//! * `no_std` support
+//!
+//!## Example
+//! For detailed example with rp-pico check [example](https://github.com/atlas-aero/rt-mcp2517/tree/main/example)
+//!
+//!## CAN TX/RX example
+//!
+//!```
+//!# use mcp2517::example::{ExampleClock,ExampleCSPin,ExampleSPIBus};
+//!# use mcp2517::can::{MCP2517,CanController};
+//!# use mcp2517::message::{Can20,TxMessage};
+//!# use mcp2517::filter::Filter;
+//!# use mcp2517::config::*;
+//!# use bytes::Bytes;
+//!# use embedded_can::{Id,StandardId};
+//!
```

Comment:
Now you've done more than necessary :). 
In the very first main example, it is then usual to show the use statements again, so that you can see at first glance what other dependencies there are.
In addition, the user can at least copy this example with one click and try it out.

```suggestion
//! use mcp2517::example::{ExampleClock,ExampleCSPin,ExampleSPIBus};
//! use mcp2517::can::{MCP2517,CanController};
//! use mcp2517::message::{Can20,TxMessage};
//! use mcp2517::filter::Filter;
//! use mcp2517::config::*;
//! use bytes::Bytes;
//! use embedded_can::{Id,StandardId};
//!
```

Reply:
Got it :+1: